src/main.c

```diff
@@ -2666,14 +2735,22 @@ static int on_config_listen_element(h2o_configurator_command_t *cmd, h2o_configu
             return -1;
         if (rcvbuf_node != NULL && h2o_configurator_scanf(cmd, *rcvbuf_node, "%u", &stream_rcvbuf) != 0)
             return -1;
+        if (reconnect_interval_node != NULL && h2o_configurator_scanf(cmd, *reconnect_interval_node, "%lu", &reconnect_interval) != 0)
+            return -1;
+        if (connections_per_thread_node != NULL && h2o_configurator_scanf(cmd, *connections_per_thread_node, "%lu", &connections_per_thread) != 0)
+            return -1;
     } break;
     default:
         h2o_configurator_errprintf(cmd, node, "value must be a string or a mapping (with keys: `port` and optionally `host`)");
         return -1;
     }
 
-    if (strcmp(type, "unix") == 0) {
+    if (strcmp(type, "reverse") != 0 && servname == NULL) {
+        h2o_configurator_errprintf(cmd, *initcwnd_node, "cannot find mandatory attribute: port");
```

Comment:
```suggestion
        h2o_configurator_errprintf(cmd, node, "type is not `reverse` and `port` is not specified");
```
2nd argument should point to the mapping within which `port` must be defined.